lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -538,7 +539,7 @@ def set_message(self, s):
 
     def set_cursor(self, cursor):
         self.canvas.get_property("window").set_cursor(cursord[cursor])
-        #self.canvas.set_cursor(cursord[cursor])
+        Gtk.main_iteration()
```

Comment:
Because otherwise the cursor is actually not set on the GUI (see docstring changed in CanvasBase.set_cursor).  Basically, what happens is

Notification that a redraw is necessary.
Redraw starts and cursor is set.
<possibly long computation>
Return to GUI thread.

Note that the GUI event loop doesn't get to run between <cursor is set> and <possibly long computation>, so depending on the details of the toolkit, the cursor change may not actually appear to the user before the end of the computation.  Letting the GUI event loop run once allows it to change the cursor.  (And upon testing, I see that Qt does not need this, probably because its version of setCursor triggers(?) the event loop.)